src/main/java/org/minions/devfund/katerinaanzoleaga/battleship/Ocean.java

```diff
@@ -0,0 +1,283 @@
+    }
+
+    /**
+     * Ocean constructor.
+     */
+    public Ocean() {
+        shotsFired = 0;
+        hitCount = 0;
+        shipsSunk = 0;
+        this.fillOcean();
+    }
+
+    /**
+     * Fills the Ocean with empty ships.
+     */
+    private void fillOcean() {
+        for (int i = 0; i < this.ships.length; ++i) {
+            for (int j = 0; j < this.ships.length; ++j) {
+                EmptySea emptySea = new EmptySea();
+                emptySea.setBowRow(i);
+                emptySea.setBowColumn(j);
+                this.ships[i][j] = emptySea;
+            }
+        }
+    }
+
+    /**
+     * Prints the ocean with the values to display the game board and ships' status.
+     *
+     */
+    public void print() {
+        String lineSeparator = System.lineSeparator();
+        StringBuilder sb = new StringBuilder();
+        Formatter fmt = new Formatter(sb);
+        int hitPosition;
+
+        final String spaces = " ";
+        for (int i = -1; i < this.ships.length; ++i) {
+            for (int j = -1; j < this.ships.length; ++j) {
+                if (i == -1 && j == -1) {
+                    fmt.format("%2s", "");
+                }
+                if (i == -1 && j != -1) {
+                    sb.append(spaces);
+                    fmt.format("%02d", j);
+                }
+                if (j == -1 && i != -1) {
+                    fmt.format("%02d", i);
+                }
+                if (i != -1 && j != -1) {
+                    hitPosition = this.ships[i][j].isInShipPosition(i, j);
+                    if (hitPosition != -1) {
+                        if (this.ships[i][j].getHit()[hitPosition]) {
+                            fmt.format("%3s", this.ships[i][j].toString());
+                        } else {
+                            fmt.format("%3s", ".");
+                        }
+                    }
+                }
+            }
+            sb.append(lineSeparator);
+        }
+        System.out.println(sb.toString());
+    }
+
+    /**
+     * Returns true if the position is valid from 0 to the length of the ocean.
+     * @param row int
+     * @param column int
+     * @return boolean
+     */
+    public boolean isValidPosition(int row, int column) {
+        int maxPosition = this.getShipArray().length - 1;
+        return row >= 0 && row <= maxPosition && column >= 0 && column <= maxPosition;
+    }
+
+    /**
+     * Verifies if a ship positon in the Ocean is 'empty'.
+     * @param row number
+     * @param column number
+     * @return Ture if the ship in this position has its type equal to 'empty'
+     */
+    public boolean isOccupied(int row, int column) {
+        return ships[row][column].getShipType() != "empty" || ships[row][column] == null;
+    }
+
+
+    /**
+     * Places the number of ships as indicated by the constant numbers in this class.
+     * It places each ship in a random position and direction.
+     */
+    public void placeAllShipsRandomly() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)